### PR TITLE
Reduce set of files copied on official builds

### PR DESCRIPTION
**Pull request by jaredpar**
This changes our official build process to no longer upload unnecessary files to cpvsbuild. Thus far I've excluded the UnitTests and PerfTests folder. We can add more as needed here but even this will significantly increase our build times and reduce storage needs.

Validation build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2040811&view=logs

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure for review

**Comment by jaredpar** [comment]
retest this please

**Comment by jaredpar** [comment]
looks like jenkins panic'd in the middle of the run

**Comment by jaredpar** [comment]
retest windows_debug_spanish_unit32_prtest  please

**Comment by jaredpar** [comment]
CC @tmat

File: .vsts-ci.yml [hunk 1]
```diff
@@ -100,15 +100,11 @@ steps:
 
 # Publish VSTS artifact that the RoslynInsertionTool is able to find by its name.
 # The backing storage of the artifact is a file share.
-- task: PublishBuildArtifacts@1
+- task: PowerShell@2
```
Comment (tmat) [inline .vsts-ci.yml]: This won't work. `PublishBuildArtifacts` not only copies files but also associates ArtifactName with the share path in VSTS artifact database. The insertion tool depends on the artifacts being available in the database.

Comment (tmat) [reply .vsts-ci.yml]: We can change the step below (`Publish Drop - VSSetup`) to PublicBuildArtifacts.

Comment (jaredpar) [reply .vsts-ci.yml]: @tmat can you elaborate a bit? I'm unsure what change I need to make here.

Comment (tmat) [reply .vsts-ci.yml]: Actually, it won't be that simple. We need to first update Roslyn insertion tool. 
Then we need to make some changes to the `Publish Drop - VSSetup` task below.

@jmarolf

**Comment by jmarolf** [comment]
> 

No `0` please

---
Refers to: 0:3 in bdbb07f. [](commit_id = bdbb07f69880a5914d486a84568b3943e4990332, deletion_comment = False)

**Comment by jmarolf** [comment]
I'll sign off on this once we have signed build and insertion working against this branch

**Comment by jaredpar** [comment]
@jmarolf 

> No 0 please or whatever that binary file is

What 0 are you referring to?

> I'll sign off on this once we have signed build and insertion working against this branch

Here is a signed build that I created. Note I didn't enable publishing though. Do I need to do that in order to validate the insertion tool works?

https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2134421&view=logs

File: .vsts-ci.yml [hunk 2]
```diff
@@ -89,17 +89,29 @@ steps:
 
 # Publish VSTS artifact that the RoslynInsertionTool is able to find by its name.
 # The backing storage of the artifact is a file share.
+#
+# The bulk of the publish work is done by the Powershell task below. This exists to create
+# the proper entries in the build log so that the insertion tool can find the drop that 
+# needs to get published
 - task: PublishBuildArtifacts@1
   displayName: Publish Drop - Binaries
```
Comment (Pilchie) [inline .vsts-ci.yml]: Rename this to `Publish Drop  - DevDivPackages`, and the one below to just `Binaries`?

**Comment by jmarolf** [comment]
> What 0 are you referring to?

image

**Comment by jmarolf** [comment]
@brettfo can you do a test insertion of this build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2134421 ?

**Comment by brettfo** [comment]
@jmarolf @jaredpar Test insertion failed.  See [this](https://dev.azure.com/devdiv/DevDiv/_releaseProgress?_a=release-environment-logs&releaseId=181663&view=mine&environmentId=688202) job for the error log.

File: .vsts-ci.yml [hunk 3]
```diff
@@ -89,17 +89,29 @@ steps:
...
     ArtifactName: '$(Build.BuildNumber)'
     publishLocation: FilePath
     TargetPath: '$(DropRoot)\Roslyn-Signed\$(Build.SourceBranchName)\$(BuildConfiguration)'
     Parallel: true
     ParallelCount: 64
   condition: succeededOrFailed()
 
+# Archive the files from the build necessary for future investigations
+- task: PowerShell@2
+  displayName: Publish Drop - Binaries Real
```
Comment (jasonmalinowski) [inline .vsts-ci.yml]: "Real" vs...fake?

**Pull request merged**